src/Scheduler.scala

```diff
@@ -0,0 +1,40 @@
+import java.util.concurrent.{Executors, PriorityBlockingQueue}
+
+import scala.concurrent.{ExecutionContext, Future}
+
+//type Task = () => Unit
+
+trait Scheduler {
+  def schedule(task: () => Unit, timestamp: Long): Unit
+  def start()(implicit ec: ExecutionContext): Unit
+}
+
+class SchedulerImp(
+    val minHeap: PriorityBlockingQueue[ScheduledTask],
+    // ExecutionContext for Future to execute a task in worker thread (not calling thread)
+    implicit val ec: ExecutionContext
+) extends Scheduler {
+
+  override def schedule(task: () => Unit, timestamp: Long): Unit = {
+    minHeap.offer(ScheduledTask(task, timestamp))
+  }
+
+  override def start()(implicit ec: ExecutionContext): Unit = {
+    Future {
```

Comment:
Using `Future` to execute this in another thread. Resolved this comment:

> this gets executed in main thread. it will block main thread further code execution once this method is called. this should be done in background thread.